cfg_mgmt/btp_application_certificate.py

```diff
@@ -0,0 +1,256 @@
+    fpath = os.path.join(temp_dir.name, fname)
+    with open(fpath, 'w', encoding='utf-8') as f:
+        f.write(content)
+    return fpath
+
+
+@dataclass
+class CertInfo:
+    id: str
+    dn: str
+    cn: str
+    serial_no: int
+
+
+class GBaasAppClient:
+    def __init__(self, auth: BtpApplicationCertificate):
+        endpoint = auth.application_endpoint()
+        self.url = f'{endpoint}/service/sps/{auth.application_id()}/apiCertificate'
+        self.clienturl = f'{endpoint}/service/sps/{auth.client_id()}/apiCertificate'
+        self.certificate_pem = auth.certificate_pem()
+        self.private_key_pem = auth.private_key_pem()
+
+    @contextmanager
+    def _session_with_cert(self):
+        temp_dir = tempfile.TemporaryDirectory()
+        session = requests.Session()
+        crt_fname = _write_temp_file(temp_dir, 'cert.pem', self.certificate_pem)
+        key_fname = _write_temp_file(temp_dir, 'key.pem', self.private_key_pem)
+        session.cert = (crt_fname, key_fname)
+        try:
+            yield session
+        finally:
+            temp_dir.cleanup()
+
+    def put_certificate(self, cert_pem: str, desc: str, scopes: list[str]) -> str:
+        idx = cert_pem.find('-----\n')
+        if idx != -1:
+            cert_pem = cert_pem[idx+6:]
+        idx = cert_pem.find('\n-----')
+        if idx != -1:
+            cert_pem = cert_pem[:idx]
+        data = {
+            'description': desc,
+            'scopes': scopes,
+            'base64': cert_pem,
+        }
+        with self._session_with_cert() as session:
+            resp = session.put(self.url, json=data)
+        resp.raise_for_status()
+        id = resp.json()['certificateId']
+        logger.info(f'Added certificate {id}')
+        return id
+
+    def delete_certificate(self, common_name: str, cert_id: str):
+        data = {
+            'certificateId': cert_id,
+        }
+        with self._session_with_cert() as session:
+            resp = session.delete(self.url, json=data)
+        resp.raise_for_status()
+        logger.info(f'Deleted certificate {common_name} ({cert_id})')
+
+    @staticmethod
+    def _find_cn(dn: str) -> str:
+        for part in dn.split(','):
+            (key,value) = part.split('=')
+            if key == 'CN':
+                return value
+        return None
+
+    def list_certificates_by_base(self, common_name_base: str) -> list[CertInfo]:
+        with self._session_with_cert() as session:
+            resp = session.get(self.clienturl)
+        resp.raise_for_status()
+        certs = []
+        for item in resp.json():
+            id = item['dnId']
+            dn = item['dn']
+            cn = GBaasAppClient._find_cn(dn)
+            if cn:
+                serial_no, base = BtpApplicationCertificate.get_serial_no_from_common_name(cn)
+                if serial_no > 0 and common_name_base == base:
+                    certs.append(CertInfo(id=id, dn=dn, cn=cn, serial_no=serial_no))
+        return certs
```

Comment:
stylistic nit: you might `yield` the values instead of creating a list (this would change the function's return type to `typing.Generator[CertInfo, None, None]`, which I consider slightly more elegant.